Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -222,13 +258,15 @@ void ScrollAnchoringController::adjustScrollPositionForAnchoring()
         invalidateAnchorElement();
         return;
     }
-    FloatSize adjustment = computeOffset(*renderBox) - m_lastOffsetForAnchorElement;
+    FloatSize adjustment = computeOffsetFromOwningScroller(*renderBox) - m_lastOffsetForAnchorElement;
     if (!adjustment.isZero()) {
-        auto newScrollPosition = frameView().scrollPosition() + IntPoint(adjustment.width(), adjustment.height());
-        LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::updateScrollPosition() for frame: " << m_frameView << " adjusting from: " << frameView().scrollPosition() << " to: " << newScrollPosition);
+        auto newScrollPosition = m_owningScrollableArea.scrollPosition() + IntPoint(adjustment.width(), adjustment.height());
+        LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::updateScrollPosition() for frame: " << frameView() << " for scroller: " << m_owningScrollableArea << " adjusting from: " << m_owningScrollableArea.scrollPosition() << " to: " << newScrollPosition);
         auto options = ScrollPositionChangeOptions::createProgrammatic();
         options.originalScrollDelta = adjustment;
-        m_frameView.setScrollPosition(newScrollPosition, options);
+        m_owningScrollableArea.setCurrentScrollType(ScrollType::Programmatic);
```

Comment:
Do we need this as well as setting the `programmatic` bit in `options`? Seems redundant.

Reply:
Also you need to save and restore the current scroll type.

Reply:
Yeah requestScrollToPosition doesn't set the current scroll type for whatever reason. Updated to restore the previous scroll type.